src/overlay/Peer.h

```diff
@@ -121,6 +145,32 @@ class Peer : public std::enable_shared_from_this<Peer>,
     uint256 mSendNonce;
     uint256 mRecvNonce;
 
+    struct OutboundSCPMessage
+    {
+        StellarMessage mMessage;
+        Peer::MessageTrackerPtr mCb;
+        VirtualClock::time_point mTimeEmplaced;
+    };
+
+    struct ReadingCapacity
+    {
+        uint32_t mFloodCapacity;
+        uint32_t mTotalCapacity;
+    };
+
+    std::vector<OutboundSCPMessage> mOutboundSCPMessages;
```

Comment:
This part kinda makes me nervous too: adding queues! A message already traverses an inbound buffered stream, the action queue, the tx queue, the action queue again, the outbound message-writing queue, and the outbound stream buffer.

I guess I feel (this is a bit abstract, please let me know why I'm wrong) like we _ought_ to be able to hook the new flow control events (stop sending / message being processed / message done / send more) into some places in the existing queues rather than adding new ones.

Reply:
(this is too abstract to be actionable -- we discussed in chat and it seems like there'd be just as much complexity moving the prioritization and dropping conditions into the action queues, so marking this resolved)